packet_pool.go

```diff
@@ -5,17 +5,68 @@ import (
 	"sync"
 )
 
-// packetPool represents a pool of packets
+// packetAccumulator keeps track of packets for a single PID and decides when to flush them
+type packetAccumulator struct {
+	dmx *Demuxer
```

Comment:
Per one of my previous comments, I feel it should be replaced with `optPacketsParser` and `programMap` directly